src/protocol/messages/mod.rs

```diff
@@ -137,7 +137,7 @@ fn read_versioned_array<R: Read, T: ReadVersionedType<R>>(
             format!("Invalid negative length for array: {}", l).into(),
         ))),
         _ => {
-            let len = len as usize;
+            let len = usize::try_from(len).map_err(ReadError::Overflow)?;
```

Comment:
Should this be `.map_err(|e| ReadVersionedError::ReadError(ReadError::Overflow(e)))?`?

Reply:
I think `thiserror` takes care of that itself:

https://github.com/influxdata/rskafka/blob/2c13f4bea7cb8ec3d0f3a5ab4eee98cf393ca50d/src/protocol/messages/mod.rs#L40-L44

See the `#[from]` macro.